ferrocene/doc/internal-procedures/src/manual-upstream-pulls.rst

```diff
@@ -10,97 +10,70 @@ What is an Upstream Pull
+request automatically. Sometimes merge conflicts may occur in which case the conflict
+needs to to be resolved manually.
 
-Manual intervention when the automation fails
----------------------------------------------
+Upstram Pull without merge conflicts
+---------------------
 
-Performing a manual pull differs slightly depending on whether the pull is done
-for the `master` or `beta` branch. The following text uses a few angle
-bracketed terms that have to be substituted as follows:
+To sign off a automatic upstream pull without merge conflicts, the reviewer only
+needs to approve the PR and comment `bors merge`. This will trigger the "full" CI
+workflow. If the workflow succeeds, the Upstream Pull is getting merged.
 
-* `<BASE_BRANCH>`: The branch of the Ferrocene repository to pull the changes into.
-  This is `main` if doing a `master` upstream pull and `release/1.70` for a
-  `beta` upstream pull.
-* `<UPSTREAM_BRANCH>`: The upstream branch to pull changes from. This is `master`
-  if doing a `master` upstream pull and `beta` for a `beta` upstream pull.
-* `<NEW_PULL_BRANCH>`: This represents the branch you will be making the pull
-  with. Replace it with a branch name of your choice, e.g.
-  `pull-upstream-master-2023-05-22`
+Upstream Pull with merge conflicts
+----------------------------------
 
-Initiating the pull
-^^^^^^^^^^^^^^^^^^^
+If an automatic upstream pull creates merge conflicts, the automation commits the
+merge conflicts and they need to be resolved manually.
 
-Run the following git commands to set up a new local `<NEW_PULL_BRANCH>`
-branch to do the pull on::
+Checkout PR
+^^^^^^^^^^^
 
-  git checkout <BASE_BRANCH> # check out the ferrocene branch to do the pull for
-  git pull --rebase # make sure your local branch is up to date
-  git submodule update # update any submodules that might've changed
-  git checkout -b <NEW_PULL_BRANCH> # checkout a new branch to do the pull with
+Start by checking out the PR. The name of the branch is
+`automation/pull-upstream/<ID>`, e.g. `automation/pull-upstream/dfvjj2s4`::
 
-Performing the pull
-^^^^^^^^^^^^^^^^^^^
+    git switch automation/pull-upstream/<ID>
 
-Now run::
+Fix conflict
+^^^^^^^^^^^^
 
-  ferrocene/tools/pull-upstream/pull.sh <UPSTREAM_BRANCH>
+There are multiple kinds of merge conflicts:
 
-The script will fetch the latest changes from `rust-lang/rust`'s
-`<UPSTREAM_BRANCH>` branch, create a commit to remove the excluded files
-(see below) from the upstream branch, and finally try to merge upstream into
-the current branch.
+General conflict
+""""""""""""""""
 
-The script is able to resolve some kinds of merge conflicts on its own. If
-there are still remaining merge conflicts after the automatic resolution the
-script will exit. You'll have to fix the merge conflicts yourself in that case:
+In order to find general conflicts search for the git conflict markers (``<<<<<<<``,
+``=======`` or ``>>>>>>>``).
```

Comment:
It's probably easier to recommend running:

```
ferrocene/ci/scripts/detect-conflict-markers.py
```

The tool automatically detects both general conflicts and deleted files (with our custom syntax).